Tools/autotest/arducopter.py

```diff
@@ -7431,6 +7431,73 @@ def PAUSE_CONTINUE(self):
 
         self.wait_disarmed()
 
+    # test pause/continue in guided mode
+    def PAUSE_CONTINUE_GUIDED(self):
+
+        destination = mavutil.location(-35.361286, 149.161107, 30, 0)
+
+        # get ready for takeoff
+        self.change_mode("GUIDED")
+        self.wait_ready_to_arm()
+        self.arm_vehicle()
+
+        # takeoff
+        self.user_takeoff(alt_min=destination.alt)
+
+        # move to a location
+        self.mav.mav.set_position_target_global_int_send(
+            0, # timestamp
+            1, # target system_id
+            1, # target component id
+            mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT_INT,
+            MAV_POS_TARGET_TYPE_MASK.POS_ONLY, # mask specifying use-only-lat-lon-alt
+            int(destination.lat * 1e7), # lat
+            int(destination.lng * 1e7), # lon
+            destination.alt, # alt
+            0, # vx
+            0, # vy
+            0, # vz
+            0, # afx
+            0, # afy
+            0, # afz
+            0, # yaw
+            0, # yawrate
+        )
+
+        # wait vehicle to move away from home
+        self.delay_sim_time(seconds_to_wait=10)
+
+        # pause guided flight
+        self.run_cmd(mavutil.mavlink.MAV_CMD_DO_PAUSE_CONTINUE,
+                     0, # param1
+                     0, # param2
+                     0, # param3
+                     0, # param4
+                     0, # param5
+                     0, # param6
+                     0 # param7
+                     )
+
+        # wait vehicle to pause
+        self.wait_groundspeed(0, 1, minimum_duration=5)
+
+        # continue guided flight
+        self.run_cmd(mavutil.mavlink.MAV_CMD_DO_PAUSE_CONTINUE,
+                     1, # param1
+                     0, # param2
+                     0, # param3
+                     0, # param4
+                     0, # param5
+                     0, # param6
+                     0 # param7
+                     )
+
```

Comment:
I'd like to see more of the different destination types tested in here, and that changing out of guided / back into guided works appropriately.